test/pipelines/unit/test_pipeline.py

```diff
@@ -24,3 +24,300 @@ def test_max_loops():
     pipe.connect("merge.value", "threshold.value")
     with pytest.raises(PipelineMaxLoops):
         pipe.run({"merge": {"value_2": 1}})
+
+
+def test_to_dict():
+    add_two = AddFixedValue(add=2)
+    add_default = AddFixedValue()
+    double = Double()
+    pipe = Pipeline(metadata={"test": "test"})
+    pipe.add_component("add_two", add_two)
+    pipe.add_component("add_default", add_default)
+    pipe.add_component("double", double)
+    pipe.connect("add_two", "double")
+    pipe.connect("double", "add_default")
+
+    res = pipe.to_dict()
+    expected = {
+        "metadata": {"test": "test"},
+        "max_loops_allowed": 100,
+        "components": {
+            "add_two": {
+                "type": "AddFixedValue",
+                "hash": id(add_two),
+                "init_parameters": {"add": 2},
+            },
+            "add_default": {
+                "type": "AddFixedValue",
+                "hash": id(add_default),
+                "init_parameters": {},
+            },
+            "double": {
+                "type": "Double",
+                "hash": id(double),
+                "init_parameters": {},
+            },
+        },
+        "connections": [
+            {"sender": "add_two.result", "receiver": "double.value"},
+            {"sender": "double.value", "receiver": "add_default.value"},
+        ],
+    }
+    assert res.keys() == expected.keys()
+    assert res["metadata"] == expected["metadata"]
+    assert res["max_loops_allowed"] == expected["max_loops_allowed"]
+    assert res["components"] == expected["components"]
+    assert res["connections"] == expected["connections"]
```

Comment:
Can't we reduce this to simply:

```suggestion
    assert res == expected
```

?

Reply:
Yeah, make sense. I've done it separately just so it's clearer what's causing eventual failures.

I'll simplify it.

Reply:
Ahh I see. I normally find pytest with `-vvv` option to be quite precise when comparing dicts.